copy/single.go

```diff
@@ -448,7 +449,13 @@ func (ic *imageCopier) copyLayers(ctx context.Context) ([]compressiontypes.Algor
 				logrus.Debugf("Skipping foreign layer %q copy to %s", cld.destInfo.Digest, ic.c.dest.Reference().Transport().Name())
 			}
 		} else {
-			cld.destInfo, cld.diffID, cld.err = ic.copyLayer(ctx, srcLayer, toEncrypt, pool, index, srcRef, manifestLayerInfos[index].EmptyLayer)
+			toc, err := chunked.GetTOCDigest(manifestLayerInfos[index].Annotations)
+			if err != nil {
+				cld.err = err
+			} else {
+				incremental := toc != nil
```

Comment:
incremental here means that only the TOC digest was validated and the layer was stored by its TOC digest, not the diffID.

When `incremental` is set we skip the diffID validation and use the digest we get from `PutBlobWithOptions`.

Reply:
Yes, _in the context of this PR_ we all know what that means.

I’m concerned that the code is clear by the time the PR is committed, so that anyone looking at it 2 years from now can maintain the required invariants for this security-critical value.

Reply:
This PR ultimately used a different approach.